### PR TITLE
CORE-19887 Re-add filtered dependency notary checks

### PR DESCRIPTION
**Pull request by nkovacsx**
### **THIS PR IS FOR 5.2.1**

### CODE REVIEW

File: notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/test/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImplTest.kt [hunk 1]
```diff
@@ -196,6 +195,8 @@ class ContractVerifyingNotaryServerFlowImplTest {
         whenever(filteredTransaction.notaryName).thenReturn(notaryServiceName)
         whenever(filteredTransaction.timeWindow).thenReturn(mockTimeWindow)
         whenever(filteredTransaction.verify()).thenAnswer {  }
+        whenever(filteredTransaction.notaryName).thenReturn(notaryServiceName)
```
Comment (jennyang-r3) [inline notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/test/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImplTest.kt:198]: this mock already exists on line 195
```suggestion
```

File: notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/test/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImplTest.kt [hunk 2]
```diff
@@ -196,6 +195,8 @@ class ContractVerifyingNotaryServerFlowImplTest {
...
+        whenever(filteredTransaction.notaryKey).thenReturn(notaryServiceCompositeKey)
```
Comment (jennyang-r3) [inline notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/test/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImplTest.kt]: Just out of curiosity, do we expect a composite key from filtered tx? In my head, it feels like it is a specific notary representation. Correct me if I'm wrong

Comment (nkovacsx) [reply notary-plugins/notary-plugin-contract-verifying/notary-plugin-contract-verifying-server/src/test/kotlin/com/r3/corda/notary/plugin/contractverifying/server/ContractVerifyingNotaryServerFlowImplTest.kt]: Good point. Maybe we should check if the notary key is composite.

**Review by jennyang-r3 — APPROVED** [review]
LGTM

**Comment by blsemo** [comment]
Duplicate of https://github.com/corda/corda-runtime-os/pull/6089

**Pull request closed without merging**